core/src/test/java/org/infinispan/persistence/BaseNonBlockingStoreTest.java

```diff
@@ -0,0 +1,660 @@
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import org.infinispan.commons.marshall.Marshaller;
+import org.infinispan.commons.marshall.ProtoStreamMarshaller;
+import org.infinispan.commons.marshall.WrappedByteArray;
+import org.infinispan.commons.marshall.WrappedBytes;
+import org.infinispan.commons.util.IntSet;
+import org.infinispan.commons.util.IntSets;
+import org.infinispan.configuration.cache.Configuration;
+import org.infinispan.configuration.cache.ConfigurationBuilder;
+import org.infinispan.container.entries.InternalCacheEntry;
+import org.infinispan.container.entries.InternalCacheValue;
+import org.infinispan.container.impl.InternalEntryFactory;
+import org.infinispan.container.impl.InternalEntryFactoryImpl;
+import org.infinispan.distribution.ch.KeyPartitioner;
+import org.infinispan.marshall.TestObjectStreamMarshaller;
+import org.infinispan.marshall.persistence.PersistenceMarshaller;
+import org.infinispan.marshall.persistence.impl.MarshalledEntryUtil;
+import org.infinispan.persistence.spi.CacheLoader;
+import org.infinispan.persistence.spi.CacheWriter;
+import org.infinispan.persistence.spi.InitializationContext;
+import org.infinispan.persistence.spi.MarshallableEntry;
+import org.infinispan.persistence.spi.NonBlockingStore;
+import org.infinispan.persistence.spi.PersistenceException;
+import org.infinispan.persistence.support.EnsureNonBlockingStore;
+import org.infinispan.persistence.support.NonBlockingStoreAdapter;
+import org.infinispan.persistence.support.WaitNonBlockingStore;
+import org.infinispan.protostream.ProtobufUtil;
+import org.infinispan.protostream.SerializationContext;
+import org.infinispan.protostream.SerializationContextInitializer;
+import org.infinispan.test.AbstractInfinispanTest;
+import org.infinispan.test.TestDataSCI;
+import org.infinispan.test.TestingUtil;
+import org.infinispan.test.data.Key;
+import org.infinispan.test.data.Person;
+import org.infinispan.test.fwk.NonBlockingTest;
+import org.infinispan.test.fwk.TestCacheManagerFactory;
+import org.infinispan.test.fwk.TestInternalCacheEntryFactory;
+import org.infinispan.util.ControlledTimeService;
+import org.infinispan.util.PersistenceMockUtil;
+import org.infinispan.util.concurrent.CompletionStages;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import io.reactivex.rxjava3.core.Flowable;
+
+/**
+ * This is a base class containing various unit tests for each and every different CacheStore implementations. If you
+ * need to add Cache/CacheManager tests that need to be run for each cache store/loader implementation, then use
+ * BaseStoreFunctionalTest.
+ */
+// this needs to be here for the test to run in an IDE
```

Comment:
Can we add this comment as part of the javadoc.

Reply:
Sorry this was just duplicated from the BaseStoreTest. Tbh I will just remove the comment :D

Reply:
I have just removed this for now, this was a copy over from the previous SPI BaseStoreTest.